prow/config/inrepoconfig_test.go

```diff
@@ -613,9 +613,9 @@ postsubmits: [{"name": "oli", "spec": {"containers": [{}]}}]`),
 
 			var p *ProwYAML
 			if headSHA == baseSHA {
-				p, err = prowYAMLGetterWithDefaults(tc.config, testGC, org+"/"+repo, baseSHA)
+				p, err = prowYAMLGetterWithDefaults(tc.config, testGC, org+"/"+repo, "main", baseSHA)
 			} else {
-				p, err = prowYAMLGetterWithDefaults(tc.config, testGC, org+"/"+repo, baseSHA, headSHA)
+				p, err = prowYAMLGetterWithDefaults(tc.config, testGC, org+"/"+repo, "main", baseSHA, headSHA)
```

Comment:
We should probably have some test cases supply the empty string to check for regressions.

Even better would be to also have a test that checks that the base branch is actually updated and that the new commits aren't pruned by subsequent `git gc` calls.

Reply:
SG. Will update.

Reply:
> We should probably have some test cases supply the empty string to check for regressions.

I've done this locally and tests seem to pass.

> Even better would be to also have a test that checks that the base branch is actually updated and that the new commits aren't pruned by subsequent `git gc` calls.

I've started looking into this but at first glance it appears a bit tricky to get right. For one thing I don't think I can use the localgit module as is.